nautobot/docs/user-guide/administration/installation/health-checks.md

```diff
@@ -0,0 +1,307 @@
+
+#### MySQL
+
+While MySQL provides the [`mysqladmin ping` CLI command](https://dev.mysql.com/doc/refman/8.0/en/mysqladmin.html), it's important to note that this command only checks whether the database server is running - it still exits with return code `0` if the server is running but not accepting connections. Therefore you might in some cases wish to run a command that actually connects to the database, such as `mysql --execute "SHOW DATABASES;`.
+
+### Redis
+
+Redis provides the [`redis-cli ping` CLI command](https://redis.io/commands/ping/) for detecting whether the Redis server is alive. It will output `PONG` on success and exit with return code `0`. Note though that it may also exit with code `0` in cases where the server has started but is not yet ready to receive or serve data.
+
+## Deployments with systemd
+
+For systemd deployments, the underlying services of PostgreSQL/MySQL and Redis integrate natively with systemd's `sd_notify` API to provide additional status information to the system, and `uwsgi` does as well. We recommend following the standard deployment patterns provided by your OS for PostgreSQL/MySQL and Redis. For the Nautobot service and Celery/Beat services, follow the Nautobot installation documentation at [Setup systemd](services.md#setup-systemd).
+
+## Kubernetes Deployments
+
+Kubernetes (k8s) distinguishes between "startup", "readiness", and "liveness" probes. In brief:
+
+- Startup probes detect whether a container has finished starting up.
+- Readiness probes detect whether a container is ready to accept traffic.
+- Liveness probes detect whether a container needs to be restarted.
+
+For more details, refer to the [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/). A worked example can be found in the [Nautobot Helm charts](https://github.com/nautobot/helm-charts) repository.
+
+### Nautobot Server Container in k8s
+
+For the Nautobot web server, you'll probably want to use `nautobot-server health_check` as a liveness probe (since it won't fail if the Nautobot server is too busy handling many HTTP requests, unlike the `/health/` endpoint) and use an HTTP request to `/health/` as a startup probe and a readiness probe (since it won't report success unless the Nautobot server is running and responding to HTTP requests). For example:
+
+```yaml
+startupProbe:
+  httpGet:
+    path: "/health/"
+    port: "http"
+  periodSeconds: 10
+  failureThreshold: 30
+
+readinessProbe:
+  httpGet:
+    path: "/health/"
+    port: "http"
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+
+livenessProbe:
+  exec:
+    command:
+      - "nautobot-server"
+      - "health_check"
+  periodSeconds: 10
+  timeoutSeconds: 10  # the CLI command takes a few seconds to start up
+  failureThreshold: 3
```

Comment:
NIT: Shall we put liveness probe to the top since it was mentioned first in the above paragraph?

Reply:
I think I was trying to keep the ordering consistent throughout the examples here. I also think logically it flows better this way - detect whether it has started up, detect whether it's ready, detect whether it's died.